packages/core/src/common/keybinding.ts

```diff
@@ -72,6 +77,78 @@ export namespace KeybindingContexts {
     };
 }
 
+export class KeybindingIndex {
+    private bindings: Keybinding[] = [];
+    private parent: KeybindingIndex;
+
+    constructor(defaultKeybindingIndex?: KeybindingIndex, shadowedKeybindings?: Keybinding[]) {
+        if (defaultKeybindingIndex) {
+            this.parent = new KeybindingIndex();
+            this.parent.bindings = defaultKeybindingIndex.bindings;
+            if (shadowedKeybindings) {
+                shadowedKeybindings.forEach(binding =>
+                    this.setKeyBinding(binding)
+                );
+            }
+        }
+    }
+
+    getKeybindingsForCommand(command: string): Keybinding[] {
+        const result: Keybinding[] = [];
+
+        if (this.parent) {
+            for (const parentBinding of this.parent.getKeybindingsForCommand(command)) {
+                if (!this.isShadowed(parentBinding)) {
+                    result.push(parentBinding);
+                }
+            }
+        }
+
+        for (const binding of this.bindings) {
+            if (binding.commandId === command) {
+                result.push(binding);
+            }
+        }
+        return result;
+    }
+
+    getKeyBindingsForKeyCode(keyCode: KeyCode): Keybinding[] {
+        const result: Keybinding[] = [];
+
+        if (this.parent) {
+            for (const parentBinding of this.parent.getKeyBindingsForKeyCode(keyCode)) {
+                if (!this.isShadowed(parentBinding)) {
+                    result.push(parentBinding);
+                }
+            }
+        }
+
+        for (const binding of this.bindings) {
+            if (KeyCode.equals(binding.keyCode, keyCode)) {
+                result.push(binding);
+            }
+        }
+        return result;
+    }
+
+    protected isShadowed(fromParent: Keybinding): boolean {
+        for (const binding of this.bindings) {
+            if (fromParent.commandId === binding.commandId
+                || KeyCode.equals(fromParent.keyCode, binding.keyCode)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    setKeyBinding(keybinding: Keybinding): void {
```

Comment:
This should support a Keybinding | string
and if it's a string use the easy keys

Reply:
I think it's not going to be called directly as the string will be parsed before that, as you really need the whole struct to set a keybinding (command id, context, keystroke) etc. Not just the keystroke I think